client/apollo/react/src/ProgressBarGroup/ProgressBarGroup.tsx

```diff
@@ -0,0 +1,58 @@
+import classNames from "classnames";
+import { ComponentProps, ComponentType, useCallback, useId } from "react";
+import { ProgressBar } from "../ProgressBar/ProgressBarCommon";
+
+const INITIAL_STEPPER_PROGRESS = 10;
+const MAX_STEPPER_PROGRESS = 100;
+
+export type ProgressBarGroupProps = {
+  currentStepProgress?: number;
+  currentStep: number;
+  nbSteps?: 3 | 4 | 5 | 6 | 7 | 8;
+  label?: string;
+  className?: string;
+  ProgressBarComponent: ComponentType<ComponentProps<typeof ProgressBar>>;
+};
+
+export const ProgressBarGroup = ({
+  currentStepProgress = INITIAL_STEPPER_PROGRESS,
+  nbSteps = 4,
+  currentStep,
+  label,
+  className,
+  ProgressBarComponent,
+}: ProgressBarGroupProps) => {
+  const stepperId = useId();
+
+  const getCurrentProgress: (index: number) => number = useCallback(
+    (index) => {
+      if (index < currentStep) {
+        return MAX_STEPPER_PROGRESS;
+      }
+      if (index === currentStep) {
+        if (currentStepProgress === 0) {
+          return INITIAL_STEPPER_PROGRESS;
+        }
+        return currentStepProgress;
+      }
+      return 0;
```

Comment:
Il est possible de simplifier un peut pour éviter l'imbrication des if.
voici une proposition
```suggestion
      if (index < currentStep) {
        return MAX_STEPPER_PROGRESS;
      }
      if (index === currentStep) {
        return currentStepProgress || INITIAL_STEPPER_PROGRESS;
      }
      return 0;
```